### PR TITLE
DOPE-217: added usage for the ignored parameters inside the Primitive type ArrayType

### PR DESCRIPTION
**Pull request by jansigi**

### CODE REVIEW

**Review by pgruntz — CHANGES_REQUESTED** [review]
Please add tests

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt [hunk 1]
```diff
@@ -12,7 +12,11 @@ import ch.ergon.dope.validtype.ValidType
 
 class Primitive<T : ValidType> : TypeExpression<T> {
     private val queryString: String
-    override fun toDopeQuery(): DopeQuery = DopeQuery(queryString = queryString, parameters = emptyMap())
+    private val parameters = mutableMapOf<String, Any>()
```
Comment (pgruntz) [inline core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt:15]: just declare the value here and create it in each constructor (like with _queryString_)

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt [hunk 2]
```diff
@@ -12,7 +12,11 @@ import ch.ergon.dope.validtype.ValidType
...
+
+    override fun toDopeQuery(): DopeQuery {
+        return DopeQuery(queryString = queryString, parameters = parameters)
```
Comment (pgruntz) [inline core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt:18]: use an expression body

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt (hunk 1 above)
Comment (pgruntz) [inline core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt:15]: If you store both, why not having only one file of type DopeQuery?

**Review by pgruntz — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt (hunk 1 above)
Comment (jansigi) [reply core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt:15]: done

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt (hunk 2 above)
Comment (jansigi) [reply core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt:18]: done

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt (hunk 1 above)
Comment (jansigi) [reply core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt:15]: done

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt [hunk 3]
```diff
@@ -11,35 +11,47 @@ import ch.ergon.dope.validtype.StringType
...
+            parameters = emptyMap(),
+        )
     }
 
     constructor(collection: Collection<TypeExpression<out ValidType>>) {
-        this.queryString = collection.joinToString(separator = ", ", prefix = "[", postfix = "]") { it.toDopeQuery().queryString }
+        val dopeQueryCollection = collection.map { it.toDopeQuery() }
+        this.dopeQuery = DopeQuery(
+            queryString = dopeQueryCollection.joinToString(separator = ", ", prefix = "[", postfix = "]") { it.queryString },
+            parameters = dopeQueryCollection.fold(emptyMap()) { parameters, field ->
```
Comment (pgruntz) [inline core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt:40]: why _field_?

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt [hunk 4]
```diff
@@ -11,35 +11,47 @@ import ch.ergon.dope.validtype.StringType
...
+                true -> TRUE.toDopeQuery().queryString
+                false -> FALSE.toDopeQuery().queryString
+            },
+            parameters = emptyMap(),
+        )
     }
 
     constructor(collection: Collection<TypeExpression<out ValidType>>) {
-        this.queryString = collection.joinToString(separator = ", ", prefix = "[", postfix = "]") { it.toDopeQuery().queryString }
+        val dopeQueryCollection = collection.map { it.toDopeQuery() }
```
Comment (pgruntz) [inline core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt:37]: would name it _dopeQueries_

**Review by pgruntz — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt (hunk 3 above)
Comment (jansigi) [reply core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt]: named it dopeQueryElement

**Review by pgruntz — APPROVED** [review]

**Pull request merged**